### PR TITLE
Migrate from json4s to circe

### PR DESCRIPTION
**Pull request by asoltysik**
This is a followup to #91 and depends on it. This PR purposefully focuses only on migrating to circe -  minimizing changes to the API, keeping all the exceptions, unsafe code etc. (I'll focus on these with #74 and #46).

### CODE REVIEW

File: src/main/scala/com.snowplowanalytics.snowplow/scalatracker/utils/JsonUtils.scala [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package com.snowplowanalytics.snowplow.scalatracker.utils
```
Comment (BenFradet) [inline src/main/scala/com.snowplowanalytics.snowplow/scalatracker/utils/JsonUtils.scala:13]: missing header

File: src/test/scala/com.snowplowanalytics.snowplow.scalatracker/StressTest.scala [hunk 2]
```diff
@@ -50,20 +52,19 @@ object StressTest {
...
-            case Some(JString(schema)) => SelfDescribingJson(schema, map("data"))
-          }
-      }
+    def parseJson(json: Json): SelfDescribingData[Json] =
+      json.asObject.map { obj =>
+        val map = obj.toMap
+        map.get("schema") match {
+          case Some(schema) => SelfDescribingData(SchemaKey.fromUri(schema.asString.get).get, map("data"))
+        }
+      }.get
```
Comment (BenFradet) [inline src/test/scala/com.snowplowanalytics.snowplow.scalatracker/StressTest.scala]: :mask:

File: src/main/scala/com.snowplowanalytics.snowplow/scalatracker/utils/JsonUtils.scala (hunk 1 above)
Comment (asoltysik) [reply src/main/scala/com.snowplowanalytics.snowplow/scalatracker/utils/JsonUtils.scala:13]: done

**Review by chuwy — APPROVED** [review]
👍

**Review by BenFradet — APPROVED** [review]

**Pull request merged**